### PR TITLE
editoast: add url crate to editoast

### LINKED ISSUE #2918: Add `url` dependency to editoast
## Description and goal

It would be nice to manipulate urls using https://crates.io/crates/url.  This is because currently we have to be super careful to check whether they end with a `/` or not, as we consider as just strings. 

Places where the type `Url` should be used instead of `String`:
1. Manipulation of url for postgres server [starts in `client/postgres_config.rs`](https://github.com/osrd-project/osrd/blob/19a77208b5a0a4da830889608cc6efec3f357590/editoast/src/client/postgres_config.rs#L28C7-L28C7)
2. Manipulation of url for redis server [starts in `client/redis_config.rs`](https://github.com/osrd-project/osrd/blob/19a77208b5a0a4da830889608cc6efec3f357590/editoast/src/client/redis_config.rs#L13)
4. In one of the `layers` views we return urls. [See here](https://github.com/osrd-project/osrd/blob/19a77208b5a0a4da830889608cc6efec3f357590/editoast/src/views/layers/mod.rs#L83)
5. There might be more, if you see more places where `String` should be `Urls`, feel free to add some changes

## Acceptance criteria
- [ ] The url crate is added as a dependency
- [ ] The modifications mentioned above are included

### PR DESCRIPTION
**Pull request by hamz2a**
closes https://github.com/osrd-project/osrd/issues/2918

### CODE REVIEW

File: editoast/src/views/layers/mod.rs [hunk 1]
```diff
@@ -203,11 +194,52 @@ mod tests {
...
                 "minzoom": 0,
                 "maxzoom": 18
             }),
-        ).await;
+        )
+        .await;
+    }
+
+    #[actix_test]
+    async fn layer_view() {
```
Comment (Castavo) [inline editoast/src/views/layers/mod.rs]: You could have used some test parametrization with `rstest`'s `case`

**Review by Castavo — APPROVED** [review]
LGTM,
tested in dev mode and with docker

File: editoast/src/views/layers/mod.rs (hunk 1 above)
Comment (hamz2a) [reply editoast/src/views/layers/mod.rs]: Done. thanks for the suggestion!

**Review by flomonster — APPROVED** [review]
LGTM

**Pull request merged**